documentation/src/main/asciidoc/user_guide/chapter-63-Streams.adoc

```diff
@@ -0,0 +1,416 @@
+              .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+----
+
+=== Distribution Cache
+
+This is where streams come into their stride.  When a stream operation is performed it will send the various intermediate and
+terminal operations to each node that has pertinent data.  This allows the resulting data to be sent back to the
+originating node and intermediate values are not required to be returned which results in much better performance.
+
+
+==== Rehash Aware
+
+Internally the data is segmented and each node only performs the operations upon the data it owns as a primary owner.  This allows
+for data to be processed evenly assuming segments are granular enough to provide for equal amounts of data on each node.
+
+When you are utilizing a distributed cache data this data can be quite volatile moving between nodes when a new node is added
+or an old one has left.  When this occurs data is shuffled between nodes.  This can cause issues with processing that data as it
+can see a subset or values a second time.
+
+Distributed Streams handle this reshuffling of data automatically so you don't
+have to worry about monitoring when nodes leave or enter.  Depending on the terminal
+operation this can be done with no progress (retry), partial progress (retry
+with less required data) or keeping track of wha keys it is has processed.
+
+It is possible but highly discouraged to disable rehash awareness on the stream.  This should only be considered if
+your request can handle only seeing a subset of data if a rehash occurs.  This can be done by invoking the
+link:https://docs.jboss.org/infinispan/{infinispanversion}/apidocs/org/infinispan/CacheStream.html#disableRehashAware--[disableRehashAware]
+method on the `CacheStream`.  The performance gain for most operations when a rehash doesn't occur is completely
+negigible.  The only exceptions are for iterator and forEach which will use less memory since they do not have
+to keep track of processed keys.
+
+WARNING: Please rethink disabling rehash awareness unless you really know what you are doing.
+
+==== Serialization / Externalization
+
+Since the operations are sent across to other nodes they must be externalizable by Infinispan marshalling.  This allows the
+operations to be sent to the other nodes.
+
+The simplest way is to cast your lambdas as Serializable as below.
+
+Unfortunately the link:https://docs.oracle.com/javase/8/docs/api/java/util/stream/Collectors.html[Collectors]
+class doesn't produce Serializable instances.  Thus if you need to use these, you can use the newly provided
+link:http://docs.jboss.org/infinispan/{infinispanversion}/apidocs/org/infinispan/stream/CacheCollectors.html[CacheCollectors]
+class which allows for a `Supplier<Collector>` to be provided.  This instance could then use the
+link:https://docs.oracle.com/javase/8/docs/api/java/util/stream/Collectors.html[Collectors]
+to supply a `Collector`
```

Comment:
I'd make this paragraph about collectors in general, not just `java.util.Collectors`. 
I'd also mention here that collectors aren't serialized between nodes, and how the results are merged on the originator.